sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageSpec.groovy

```diff
@@ -1,10 +1,84 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.storage.common.test.shared;
+package com.azure.storage.common.test.shared
 
-import spock.lang.Specification;
+import com.azure.core.http.HttpClient
+import com.azure.core.http.ProxyOptions
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder
+import com.azure.core.http.policy.HttpPipelinePolicy
+import com.azure.core.test.InterceptorManager
+import com.azure.core.test.TestMode
+import com.azure.core.util.Configuration
+import spock.lang.Specification
 
 class StorageSpec extends Specification {
     protected static final TestEnvironment ENVIRONMENT = new TestEnvironment();
+
+    private String testName
+    private InterceptorManager interceptorManager
+    private StorageResourceNamer resourceNamer
+
+    def setup() {
+        testName = getTestName()
+        if (shouldUseThisToRecord()) {
+            interceptorManager = new InterceptorManager(testName, ENVIRONMENT.testMode)
+            resourceNamer = new StorageResourceNamer(testName, ENVIRONMENT.testMode, interceptorManager.getRecordedData())
+        }
+        System.out.printf("========================= %s =========================%n", testName)
+    }
+
+    def cleanup() {
+        if (shouldUseThisToRecord()) {
+            interceptorManager.close()
+        }
+    }
+
+    // TODO (kasobol-msft) Remove this when all modules are migrated.
+    protected shouldUseThisToRecord() {
```

Comment:
This will be removed in couple of days, so I won't be inventing better way to do this nor naming :-)